lib/runtime/extrinsic/unchecked_extrinsic.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2020 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+package extrinsic
+
+import (
+	"math/big"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto"
+	"github.com/ChainSafe/gossamer/lib/scale"
+)
+
+// TODO determine how to get these values from the Runtime
+const specVersion uint32 = 193 // encoded as additional singed data when building UncheckedExtrinsic
+//const transactionVersion uint32 = 1 // encoded as additional singed data when building UncheckedExtrinsic
+
+// Call interface for method extrinsic is calling
+type Call byte
+
+// consts for node_runtime calls
+const (
+	System Call = iota
+	Utility
+	Babe
+	Timestamp
+	Authorship
+	Indices
+	Balances
+	Staking
+	Session
+)
+
+// Pallet for runtime sub-calls
+type Pallet int
+
+// pallet_balances calls
+const (
+	PB_Transfer Pallet = iota
+	PB_Set_balance
+	PB_Force_transfer
+	PB_Transfer_keep_alive
+)
+
+// pallet_system calls
+const (
+	SYS_fill_block Pallet = iota
+	SYS_remark
+	SYS_set_heap_pages
+	SYS_set_code
+	SYS_set_storage
+	SYS_kill_storage
+	SYS_kill_prefix
+)
+
+// session calls
+const (
+	SESS_set_keys Pallet = iota
+)
+
+// Function struct to represent extrinsic call function
+type Function struct {
+	Call     Call
+	Pallet   Pallet
+	CallData interface{}
```

Comment:
is there a reason that `Call` comes before `Pallet`? I thought that the first byte was the palette to call followed by the function in that pallet, but could be wrong

Reply:
I think your right, I think I'm maybe using poor names, because it is the Pallet, then the function within the Pallet.

Reply:
ah maybe update the naming then? it looks like it's the other way around